### ISSUE TITLE
Error 500 when processing SMS message missing year from Bikram Sambat aggregate

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#7933]
**Describe the bug**
If a form definition contains a bikram sambat aggregate field, but the receiving SMS is missing this field, API will throw an error and not process the record. 

**To Reproduce**
1. Create a form with a `bsAggregate` field (added with issue: https://github.com/medic/cht-core/issues/4613) 
2. Create a record that is missing this field.
3. See error

**Expected behavior**
We should be able to gracefully handle missing fields in SMS forms. 

**Logs**
```
ERROR: Can not aggregate bsAggreDate without bsYear 
   [stack]: 'TypeError: Cannot read property \'patient_id\' of undefined\n    at Object.exports.merge (/srv/software/medic-api/md5-IaUJW7p4rpaMwbWhKl5C1A==/src/services/report/smsparser.js:373:35)\n    at getDataRecord (/srv/software/medic-api/md5-IaUJW7p4rpaMwbWhKl5C1A==/src/services/records.js:129:17)\n    at Object.createByForm (/srv/software/medic-api/md5-IaUJW7p4rpaMwbWhKl5C1A==/src/services/records.js:190:10)\n    at messages.map.message (/srv/software/medic-api/md5-IaUJW7p4rpaMwbWhKl5C1A==/src/services/messaging.js:174:48)\n    at Array.map (<anonymous>)\n    at createDocs (/srv/software/medic-api/md5-IaUJW7p4rpaMwbWhKl5C1A==/src/services/messaging.js:174:25)\n    at validateIncomingMessages.then.messages (/srv/software/medic-api/md5-IaUJW7p4rpaMwbWhKl5C1A==/src/services/messaging.js:233:25)\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:188:7)',
RES POST /api/sms HTTP/1.1 500 35 134.816 ms
ERROR: Server error: { TypeError: Cannot read property 'patient_id' of undefined
```

**Environment**
- Instance: production instance
- App: api
- Version: 3.15+

**Additional context**
Gateway will retry sending the same SMS to API over and over. 
Workaround is to remove form definitions that use this field type.

**Comment by dianabarsan** [comment]
Backporting this to `3.17.x` and `3.16.x`

**Comment by dianabarsan** [comment]
This is ready for AT on https://github.com/medic/cht-core/pull/7934

To AT, create a form that has a `bsAggregate` date, for example: https://github.com/medic/cht-core/blob/7592-users-have-multiple-roles/api/tests/form-definitions.js#L715 . Then submit this form, omitting the bs date, through an SMS endpoint, using textform format. 

The PR contains an example of a call that should test the fix working: https://github.com/medic/cht-core/pull/7934/files#diff-181bd8cbdd40299fda246b737db4b932b32680d0cf852af1c58e57e5bb87456fR650

**Comment by tatilepizs** [comment]
Thanks @dianabarsan for the details and examples, it was really useful.

### Test Details:

**Config:** Default
**Environment:** Local
**Platform:** WebApp
**Browser:** Chrome

---

### Test scenario:
Using this [form example]( https://github.com/medic/cht-core/blob/7592-users-have-multiple-roles/api/tests/form-definitions.js#L715), and [this documentation](https://docs.communityhealthtoolkit.org/apps/tutorials/sms-forms/) I uploaded the form to test it through SMS.

### Reproducible on `Master`
Using this test message `YYYS 12345` the report was not sent and an error was shown.

<details>
<summary>Images attached.</summary>

![image](https://user-images.githubusercontent.com/94494491/204409410-5286f91f-7751-473d-971e-dbf8e4ad962c.png)
![image](https://user-images.githubusercontent.com/94494491/204409356-6292bd03-11c8-4c85-899a-04664ab92f40.png)

</details>

### Fixed on `7933-fix-bsaggregate-error`
Using this test message `YYYS 12345` the error is still present in the logs, but the report was sent and the error is shown in a better way, with more details for the user.

<details>
<summary>Images attached.</summary>

![image](https://user-images.githubusercontent.com/94494491/204409816-ede9a456-b553-4503-b50a-2e3a9ad20170.png)
![image](https://user-images.githubusercontent.com/94494491/204409786-27218789-f074-4e38-a936-b3553a89f9c1.png)
![image](https://user-images.githubusercontent.com/94494491/204409881-dcbc06d2-1598-49f3-a172-e0802a1640d6.png)

</details>

**Comment by dianabarsan** [comment]
Merged to `master` and backported to `4.0.x`, `3.17.x` and `3.16.x`.
